src/userdev/java/net/minecraftforge/gradle/userdev/UserDevPlugin.java

```diff
@@ -251,4 +257,141 @@ private void createRunConfigsTasks(@Nonnull Project project, ExtractNatives extr
+            {
+                validInformation = information;
+            }
+            else if (!validInformation.equals(information))
+            {
+                throw new IllegalStateException(String.format(
+                  "Found multiple minecraft dependencies, with different dependency information. This is not allowed. The entire project has to depend on one minecraft version: %s",
+                  validInformation));
+            }
+
+            validConfigurations.add(configuration);
+        }
+
+        if (validConfigurations.isEmpty())
+            throw new IllegalStateException("At least one configuration has to depend on minecraft.");
+
+        return validConfigurations;
+    }
+
+    /**
+     * Extracts the minecraft dependency information string from the configuration.
+     * If no dependency is found an empty string is returned.
+     *
+     * There needs to be exactly one, otherwise an exception is thrown.
+     * The minecraft dependency needs to be a maven dependency.
+     *
+     * @param configuration The configuration to extract the information from.
+     * @return A string containing the maven dependency information for minecraft, or an empty one, if no dependency is found.
+     * @throws IllegalStateException thrown when the configuration is not set up correctly.
+     */
+    private String getMinecraftDependencyInformationFromConfiguration(Configuration configuration) throws IllegalStateException
+    {
+        if (configuration.getDependencies().size() == 0)
+            return "";
+
+        if (configuration.getDependencies().size() > 1)
+            throw new IllegalStateException(String.format("Configuration: %s contains more then one minecraft dependency", configuration.getName()));
+
+        final Dependency minecraftDependency = new ArrayList<>(configuration.getDependencies()).get(0);
+        if (!(minecraftDependency instanceof ExternalModuleDependency))
+            throw new IllegalArgumentException("Minecraft dependency: " + minecraftDependency.toString() + " of configuration: " + configuration.getName() + " must be a maven dependency.");
+
+        return String.format("%s:%s:%s", minecraftDependency.getGroup(), minecraftDependency.getName(), minecraftDependency.getVersion());
+    }
+
+    /**
+     * Generates a minecraft user repo from the project and valid configurations.
+     *
+     * @param project The project to generate for.
+     * @param extension The extensions with the project information.
+     * @param validMinecraftConfigurations The valid configurations.
+     * @return The minecraft user repo.
+     */
+    private MinecraftUserRepo generateMinecraftUserRepo(Project project, UserDevExtension extension, List<Configuration> validMinecraftConfigurations)
+    {
+        //We can do this since the configurations have been validated.
+        final Dependency dependency = new ArrayList<>(validMinecraftConfigurations.get(0).getDependencies()).get(0);
```

Comment:
Is there something that prevents them from setting the wrong value for the different configurations?
This just picks the first and goes with it. The first is whatever happens to be in their build.gradle.
What is the benefit of this, over being able to just have configurations.api.extendsFrom(configurations.minecraft) ?